### PR TITLE
fix attr cop implementation bbatsov/rubocop#2483

### LINKED ISSUE #2483: rubocop breaks code when "fixing" `attr :name, true`
Rubocop replaces

`attr :name, true`

with 

`attr_reader :name, true`

which is plain wrong. If it would replace that line at all, it should use

`attr_writer :name`

instead.

### PR DESCRIPTION
**Pull request by weh**
extend the attr cop implementation to respect the additional boolean param.
Fixes #2483

### CODE REVIEW

**Comment by alexdowad** [comment]
Looks good! I would suggest that you squash these 3 commits into a single commit. Do you know how to do this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
 * [#2452](https://github.com/bbatsov/rubocop/issues/2452): `Style/TrailingComma` incorrectly categorizes single-line hashes in methods calls. ([@panthomakos][])
 * [#2441](https://github.com/bbatsov/rubocop/issues/2441): `Style/AlignParameters` doesn't crash if it finds nested offenses. ([@alexdowad][])
 * [#2436](https://github.com/bbatsov/rubocop/issues/2436): `Style/SpaceInsideHashLiteralBraces` doesn't mangle a hash literal which is not surrounded by curly braces, but has another hash literal which does as its first key. ([@alexdowad][])
+* [#2483](https://github.com/bbatsov/rubocop/issues/2483): Bugfix: Differentiate between attr_accessor and attr_reader. ([@weh][])
```
Comment (bbatsov) [inline CHANGELOG.md]: Remove the word "Bugfix", as it's implied from the section this is listed in.

File: lib/rubocop/cop/style/attr.rb [hunk 2]
```diff
@@ -14,7 +14,24 @@ def on_send(node)
         end
 
         def autocorrect(node)
-          ->(corrector) { corrector.replace(node.loc.selector, 'attr_reader') }
+          _receiver, _method_name, attr_name, setter = *node
+          node_expr = node.loc.expression
+          attr_expr = attr_name.loc.expression
+          replacement = 'attr_reader'
```
Comment (bbatsov) [inline lib/rubocop/cop/style/attr.rb]: Changing this "default value" afterwards seems a bit confusing to me. I'd rather drop the default.

File: lib/rubocop/cop/style/attr.rb [hunk 3]
```diff
@@ -14,7 +14,24 @@ def on_send(node)
...
+
+          if setter
```
Comment (bbatsov) [inline lib/rubocop/cop/style/attr.rb]: I'd merge the two `if`s.

**Comment by bbatsov** [comment]
Update the commit message to match our style - `[Fix #...] Differentiate`...

File: CHANGELOG.md (hunk 1 above)
Comment (bbatsov) [inline CHANGELOG.md]: You should also mention the name of the cop, which is affected by the bugfix as other changelog entries do.

**Comment by weh** [comment]
i merged the two if s. don't know why i did not see that before.

the "default value" thing was introduced to avoid a too long line (87).
i think the changed version should be clearer now.

can you explain me why the build failed? 
https://travis-ci.org/bbatsov/rubocop/jobs/96621263

**Comment by weh** [comment]
@bbatsov do you know why the build failed?

**Comment by alexdowad** [comment]
> do you know why the build failed?

There's a bug in the new version of the `bundler` gem, which has broken the build for every single Ruby project which runs tests on JRuby.

**Comment by bbatsov** [comment]
The commit message should still be updated.

**Comment by bbatsov** [comment]
:+1:

**Comment by alexdowad** [comment]
Great job, @weh!

**Pull request merged**